src/components/FilterBar.jsx

```diff
@@ -37,14 +37,34 @@ const sortingOptions = {
   amountHigh: 'Amount (High to Low)',
 };
 
+const sortParams = {
+  deadlineSoon: {
+    field: 'deadline',
+    dir: 'asc',
+  },
+  deadlineLatest: {
+    field: 'deadline',
+    dir: 'desc',
+  },
+  amountLow: {
+    field: 'amount.min',
+    dir: 'asc',
+  },
+  amountHigh: {
+    field: 'amount.max',
+    dir: 'desc',
+  },
+};
+
 export default function FilterBar(props) {
-  const { setSort, setQueryParam, queryParams } = props;
+  const { setQueryParam, queryParams } = props;
   const classes = useStyles();
   const { minAmount, maxAmount } = queryParams;
 
   function updateSortingParams(sortOption) {
-    if (sortOption in sortingOptions) {
-      setSort(sortOption);
+    if (sortOption in sortParams) {
+      setQueryParam('sortDir', sortParams[sortOption].dir);
+      setQueryParam('sortField', sortParams[sortOption].field);
```

Comment:
Not sure why it only updates and displays on the URL which ever is last? Starting to get frustrating as it should work

Reply:
Hmm, that's because you're always calling `setQueryParam('sortDir', sortParams[sortOption].dir)`, which always calls `history.push` and refreshes the component before it can ever set `sortField`. I think this can be fixed by only calling `history.push` if it's necessary in `setQueryParam`. I'll leave some code suggestions in `Scholarships.jsx` that should fix this.